### ISSUE TITLE
EPG not updating past 10/27 except for 1 channel

### ISSUE THREAD

**Issue by squirtbrnr** [edit4ever/script.module.zap2epg#45]
My EPG data is not updating.  It was working fine for the longest time.  I fetch 2 weeks of EPG data, however lately every channel was able to fetch up to 10/27, then after that there's nothing.  Except for one channel, I have data for that channel out the full 2 weeks.  I verified Zap2it has EPG data that far out for all of the channels.  It seems either something is not being parsed or is malformed and screwing up the EPG.

Let me know what logs or data is needed.

**Comment by squirtbrnr** [comment]
@Kensit looks like that is the case. Here’s a screenshot of that episode epg in tvheadend. And that show is on the one channel that actually has the full 2 weeks of epg, but all other stations show blank.

**Comment by jbreen60030** [comment]
Actually, to solve for my older version (0.7.4 Python 2) I included in the "include block"

```
import cgi 
from cgi import escape
```
and then everywhere that the `re.sub('&','&amp',x)` existed, I changed it to 
`x = cgi.escape(x, quote=true)` 

This will handle the `<. > , &` along with any other HTML chars that need to be escaped including the double quote. 
I also, for titles got rid of the escaping of the double quote char otherwise you end up with ` \&amp` which just looks funny.
e.g. 
```  
xthisTitle = edict['eptitle']
xthisTitle = xthisTitle.replace('\\\"','\"')
fh.write('\t\t<sub-title lang=\"'+ lang + '\">' + cgi.escape(xthisTitle, quote=True ) + '</sub-title>\n')
```
